envoy/datadog_checks/envoy/envoy.py

```diff
@@ -159,7 +163,24 @@ def _collect_metadata(self, stats_url):
             #   "state": "LIVE",
             #   ...
             # }
-            raw_version = response.json()["version"].split('/')[1]
+            try:
+                raw_version = response.json()["version"].split('/')[1]
+            except json.decoder.JSONDecodeError as e:
+                self.log.debug('Error decoding json for url=`%s`. Error: %s', server_info_url, str(e))
+
+            if raw_version is None:
+                # Search version in server info for Envoy version <= 1.8
+                # Example:
+                #     envoy 5d25f466c3410c0dfa735d7d4358beb76b2da507/1.8.0/Clean/RELEASE live 581130 581130 0
+                content = response.content.decode()
+                found = LEGACY_VERSION_RE.search(content)
```

Comment:
I think splitting by `/` is safer as it might not always be x.y.z

Reply:
I don’t think that’s the case (not 100% sure though). The fix is for handling an old format. Splitting by ‘/’ might not be safe for it’s own reasons (e.g. the doc indicate `envoy 267724/RELEASE live 1571 1571 0`).

https://www.envoyproxy.io/docs/envoy/v1.8.0/operations/admin#get--server_info